packages/dd-trace/src/appsec/iast/analyzers/weak-hash-analyzer.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+const { storage } = require('../../../../../datadog-core')
+const { IAST_CONTEXT_KEY } = require('./../index')
+const Analyzer = require('./vulnerability-analyzer')
+
+const INSECURE_HASH_ALGORITHMS = [
+  'md4', 'md4withrsaencryption', 'rsa-md4',
+  'rsa-md5', 'md5', 'md5-sha1', 'ssl3-md5', 'md5withrsaencryption',
+  'rsa-sha1', 'rsa-sha1-2', 'sha1', 'md5-sha1', 'sha1withrsaencryption', 'ssl3-sha1'
+]
+
+class WeakHashAnalyzer extends Analyzer {
+  constructor () {
+    super('WEAK_HASH_ANALYZER')
+    this.addSub('asm:crypto:hashing:start', this._handler)
```

Comment:
be aware with `this._handler`, it returns a method, not an object, you don't know what will be the `this` for it.
```suggestion
this.addSub('asm:crypto:hashing:start', (data) => this._handler(data))
```